### DISCUSSION TITLE
make method static if its called from static method

### DISCUSSION THREAD

**Discussion by themark147** [rectorphp/rector#8198] [Q&A]
hello community is there any rule for making methods static if they are called from static method

example input:
```
final class DemoFile
{
    public static function run()
    {
        $this->a();
    }
    
    public function a()
    {
        return 1;
    }
}
```

expected output
```
final class DemoFile
{
    public static function run()
    {
        self::a();
    }
    
    public static function a()
    {
        return 1;
    }
}
```

its because I migrated unit tests from 9 to 10 but I had non static calls of methods in data providers. And after migration data providers  became static

thanks a lot

**Comment by samsonasik** [comment]
Hi, you may looking for `StaticDataProviderClassMethodRector` see https://github.com/rectorphp/rector-phpunit/blob/main/docs/rector_rules_overview.md#staticdataproviderclassmethodrector

**Reply by themark147** [reply]
nope cause I already have dataProviders as static method but in these providers Im callin others methods which are NON static and I need to rewrite them as static
```
public static function provideData()
     {
         yield [$this->random()]; <-- and this have to be rewrite like this: self::random()
     }
public function random(): int <-- and this as public STATIC function ...
{
    return 1;
}
```

**Reply by samsonasik** [reply]
I see, could you create failing fixture PR for it to https://github.com/rectorphp/rector-phpunit/tree/main/rules-tests/PHPUnit100/Rector/Class_/StaticDataProviderClassMethodRector/Fixture ? Thank you.

**Reply by themark147** [reply]
sure I will try :))

**Reply by themark147** [reply]
here it is @samsonasik https://github.com/rectorphp/rector-phpunit/pull/255 :blush:
